File: server/internal/server/response.go
package server

type EmbedField struct {
	Name   string `json:"name"`
	Value  string `json:"value"`
	Inline bool   `json:"inline"`
}

type Embed struct {
	Title       string       `json:"title"`
	Description string       `json:"description"`
	Fields      []EmbedField `json:"fields,omitempty"`
}

type ReturnData struct {
	Embeds  []Embed `json:"embeds,omitempty"`
	Content string  `json:"content,omitempty"`
}

File: client/windiafaq/tcp/response.py
import discord

from windiafaq.discord.embed import NormalEmbed, EmbedField


class ServerResponseError(Exception):
    """Raised when the server has encountered an exception handling a command"""


class Response(dict):
    def __init__(self, **kwargs):
        if error := kwargs.pop("error", None):
            raise ServerResponseError(error)
            
        super().__init__(**kwargs)

    def embeds(self) -> list[discord.Embed] | None:
        if not (embeds := self.pop("embeds", None)):
            return None

        _embeds = []

        for embed in embeds:
            _embed = NormalEmbed(title=embed["title"], description=embed["description"])
            if fields := embed.pop("fields"):
                for field in fields:
                    _embed.add_field(EmbedField(name=field["name"], value=field["value"], inline=field["inline"]))

            _embeds.append(_embed)

        return _embeds

    @property
    def content(self) -> str | None:
        return self.pop("content", None)
